packages/admin/admin/src/form/FinalFormFileSelect.tsx

```diff
@@ -0,0 +1,273 @@
+            cursor: "pointer",
+            "&$droppableAreaIsDisabled": {
+                cursor: "default",
+            },
+            "&$droppableAreaHasError": {
+                border: `1px dashed ${palette.error.main}`,
+            },
+        },
+        droppableAreaCaption: {
+            padding: "30px",
+            color: palette.grey[400],
+        },
+        droppableAreaError: {
+            position: "absolute",
+            top: 0,
+            right: 0,
+            padding: "10px",
+            color: palette.error.main,
+        },
+        fileList: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            gap: "2px",
+            width: "100%",
+        },
+        fileListItem: {
+            display: "flex",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            borderRadius: "4px",
+            background: palette.grey[50],
+            justifyContent: "space-between",
+            width: "100%",
+            boxSizing: "border-box",
+        },
+        fileListItemInfos: {
+            display: "flex",
+            justifyContent: "end",
+            gap: "10px",
+        },
+        rejectedFileListItem: {
+            display: "flex",
+            padding: "14px 15px",
+            alignItems: "center",
+            gap: "10px",
+            borderRadius: "4px",
+            background: palette.grey[50],
+            justifyContent: "space-between",
+            border: `1px dashed ${palette.error.main}`,
+            color: palette.error.main,
+            width: "100%",
+            boxSizing: "border-box",
+        },
+        errorMessage: {
+            display: "flex",
+            alignItems: "center",
+            color: palette.error.main,
+            gap: "5px",
+        },
+        droppableAreaIsDisabled: {},
+        droppableAreaHasError: {},
+        fileListText: {
+            textOverflow: "ellipsis",
+            overflow: "hidden",
+            whiteSpace: "nowrap",
+        },
+    });
+};
+
+export interface FinalFormFileSelectProps extends FieldRenderProps<File | File[], HTMLInputElement> {
+    disableDropzone?: boolean;
+    disableSelectFileButton?: boolean;
+    accept?: Accept;
+    maxSize?: number;
+    maxFiles?: number;
+    iconMapping?: {
+        delete?: React.ReactNode;
+        info?: React.ReactNode;
+        select?: React.ReactNode;
+    };
+}
+
+const FinalFormFileSelectComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormFileSelectProps> = ({
+    classes,
+    disabled,
+    disableDropzone,
+    disableSelectFileButton,
+    accept,
+    maxSize = 50 * 1024 * 1024,
+    maxFiles,
+    input: { onChange, value: fieldValue, multiple: multipleFiles },
+    iconMapping = {},
+}) => {
+    const { delete: deleteIcon = <Delete />, info: infoIcon = <Error color="error" />, select: selectIcon = <Select /> } = iconMapping;
```

Comment:
An icon's name should make it easy to understand what it is used for. 
I think `info` is a bit confusing. 

```suggestion
    const { delete: deleteIcon = <Delete />, error: errorIcon = <Error color="error" />, select: selectIcon = <Select /> } = iconMapping;
```